packages/core-components/src/lib/unsorted/viz/Column.svelte

```diff
@@ -56,35 +56,61 @@
 
 	// Formatting:
 	export let fmt = undefined;
-
-	// Delta:
 	export let downIsGood = false;
-	downIsGood = downIsGood === 'true' || downIsGood === true;
 	export let showValue = true;
-	showValue = showValue === 'true' || showValue === true;
-
 	export let deltaSymbol = true;
-	deltaSymbol = deltaSymbol === 'true' || deltaSymbol === true;
 
-	let options = {
-		id: id,
-		title: title,
-		align: align,
-		wrap: wrap,
-		contentType: contentType,
-		height: height,
-		width: width,
-		alt: alt,
-		openInNewTab: openInNewTab,
-		linkLabel: linkLabel,
-		fmt: fmt,
-		downIsGood: downIsGood,
-		deltaSymbol: deltaSymbol,
-		showValue: showValue
+	const getOptions = () => {
+		downIsGood = downIsGood === 'true' || downIsGood === true;
+		showValue = showValue === 'true' || showValue === true;
+		deltaSymbol = deltaSymbol === 'true' || deltaSymbol === true;
+		return {
+			id: id,
+			title: title,
+			align: align,
+			wrap: wrap,
+			contentType: contentType,
+			height: height,
+			width: width,
+			alt: alt,
+			openInNewTab: openInNewTab,
+			linkLabel: linkLabel,
+			fmt: fmt,
+			downIsGood: downIsGood,
+			deltaSymbol: deltaSymbol,
+			showValue: showValue
+		};
 	};
 
-	props.update((d) => {
-		d.columns.push(options);
-		return d;
-	});
+	const updateProps = () => {
+		const options = getOptions();
+		props.update((d) => {
+			const matchingIndex = d.columns.findIndex((c) => c.id === id);
+			if (matchingIndex === -1) {
+				d.columns.push(options);
+			} else {
+				d.columns = [
+					...d.columns.slice(0, matchingIndex),
+					options,
+					...d.columns.slice(matchingIndex + 1)
+				];
+			}
+			return d;
+		});
+	};
+	$: id,
+		title,
```

Comment:
Is there a more succinct way of effectively indicating "all props" here?

Reply:
You may be able to reference `$$props`, but I'm not sure if that's reactive

Reply:
After some quick testing, getting unexpected results and bad performance. I think it gets a new reference every time even if the specific constituent props didn't change, which would cause the columns to constantly rerender.

Reply:
Ah yeah that's not ideal

Options could be declared this way potentially:

```javascript
$: options = {
    id, title, // ...
}
```

Reply:
Awesome! I'm glad that worked